File: go/test/protocol/test_case_duration.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"time"
)

type TestCaseDuration struct {
	Hours        int `yaml:"hours" toml:"hours"`
	Minutes      int `yaml:"minutes" toml:"minutes"`
	Seconds      int `yaml:"seconds" toml:"seconds"`
	Milliseconds int `yaml:"milliseconds" toml:"milliseconds"`
	Microseconds int `yaml:"microseconds" toml:"microseconds"`
}

func (tcd TestCaseDuration) ToDuration() time.Duration {
	return time.Duration(
		tcd.Hours,
	)*time.Hour + time.Duration(
		tcd.Minutes,
	)*time.Minute + time.Duration(
		tcd.Seconds,
	)*time.Second + time.Duration(
		tcd.Milliseconds,
	)*time.Millisecond + time.Duration(
		tcd.Microseconds,
	)*time.Microsecond
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseDuration.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseDuration
    {
        public int Hours { get; set; }

        public int Minutes { get; set; }

        public int Seconds { get; set; }

        public int Milliseconds { get; set; }

        public int Microseconds { get; set; }

        public TimeSpan ToTimeSpan()
        {
            return new TimeSpan(0, Hours, Minutes, Seconds, Milliseconds, Microseconds);
        }
    }
}
